netmiko/nokia/nokia_srl.py

```diff
@@ -0,0 +1,197 @@
+
+    Not applicable in Nokia SRL:
+        - check_enable_mode()
+        - enable()
+        - exit_enable_mode()
+
+    Overriden methods to adapt Nokia SR OS behavior (changed):
+        - session_preparation()
+        - set_base_prompt()
+        - config_mode()
+        - exit_config_mode()
+        - check_config_mode()
+        - save_config()
+        - commit()
+        - strip_prompt()
+        - enable()
+        - check_enable_mode()
+
+    By default, the SR Linux CLI prompt consists of two lines of text,
+    indicating with an asterisk whether the configuration has been modified or
+    a plus sign whether the configuration has been saved, the current mode and
+    session type, the current CLI context, and the host name of the SR Linux device.
+
+    Examples:
+
+    --{ running }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ * candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    --{ + candidate private private-admin }--[ interface ethernet-1/1 subinterface 1 ]--
+    A:ams01#
+
+    This class support the default prompt configuration.
+    """
+
+    def session_preparation(self) -> None:
+        self._test_channel_read()
+        self.ansi_escape_codes = True
+        # Bottom toolbar text not required
+        command = "environment cli-engine type basic"
+        self.write_channel(self.normalize_cmd(command))
+        command = "environment complete-on-space false"
+        self.write_channel(self.normalize_cmd(command))
+        self.set_base_prompt()
+        time.sleep(10 * self.global_delay_factor)
+        self.clear_buffer()
+
+    def set_base_prompt(
+        self,
+        pri_prompt_terminator: str = "#",
+        alt_prompt_terminator: str = "",
+        delay_factor: float = 1.0,
+        pattern: Optional[str] = r"(\n\-\-\{.+\}\-\-\[.*\]\-\-\n.*# )",
+    ) -> str:
+        return super().set_base_prompt(
+            pri_prompt_terminator, alt_prompt_terminator, delay_factor, pattern
+        )
+
+    def config_mode(
+        self,
+        config_command: str = "enter candidate private",
+        pattern: str = "]--",
+        re_flags: int = 0,
+    ) -> str:
+
+        output = super().config_mode(
+            config_command=config_command, pattern="", re_flags=re_flags
+        )
+        return output
+
+    def check_config_mode(
+        self,
+        check_string: str = r"\n--{( | \* | \+ | \+\* | \!\+ | \!\* )candidate",
+        pattern: str = " ]--",
+    ) -> bool:
+        self.write_channel(self.RETURN)
+        """
+        supported first line prompt configuration:
+        {modified_flags}{mode_and_session} }}--[ {pwc} ]
+        """
+        if not pattern:
+            output = self.read_channel_timing(read_timeout=10.0)
+        else:
+            output = self.read_until_pattern(pattern=pattern)
+
+        matches = re.search(check_string, output)
+        return True if matches else False
```

Comment:
Let's actually add some sort of argument to the parent method (as I had been thinking about doing this anyways). Something like,

force_regex = True

And in the `check_config_mode()` parent method in base_connection.py we will do:

```python
     def check_config_mode(
         self, 
         check_string: str = "", 
         pattern: str = "",
         force_regex: bool = False
) -> bool:
```

In other words, in the `check_config_mode()` parent method we will provide an option for this to be `regex-based` which we will default to False for backwards compatibility.

And then in your code we will just call `super()` on it with force_regex=True.